src/index.ts

```diff
@@ -94,13 +103,19 @@ function checkDiskSpace(directoryPath: string, dependencies: Dependencies = {
 	 *
 	 * @param directoryPath - The file/folder path from where we want to know disk space
 	 */
-	function checkWin32(directoryPath: string): Promise<DiskSpace> {
+	async function checkWin32(directoryPath: string): Promise<DiskSpace> {
 		if (directoryPath.charAt(1) !== ':') {
 			return new Promise((resolve, reject) => {
 				reject(new InvalidPathError(`The following path is invalid (should be X:\\...): ${directoryPath}`))
 			})
 		}
 
+		const cpExecFileAsync = promisify(dependencies.cpExecFile)
+
+		// Some users have WMI disbled, so we attempt to run it before check
+		await cpExecFileAsync('net', ['start', 'winmgmt']).catch(() => undefined)
```

Comment:
This only works if launched as admin, I do not think it can work in many apps launched as a normal user.

Also, I think we can make that line run once, then skip it for the next calls.

So I'm wondering if we should keep that in the code, or move it to another function that will be called by the app code if needed. What did you mean about that?

Moreover, from what I've read, it seems that the service automatically starts on the first call if it was not already running. So, I did not think this fix the issue if the service is disabled. Did you test it?

Reply:
Actually I expected to run it as admin, but I understand your concerns that many apps depending on this library doesn't want to deal with UAC stuffs.

I didn't know `wmic` calls automatically starts the service by default, and I think this line can be deleted since `wmic` already does the job. I'll make some changes to address this.